src/util/string.h

```diff
@@ -235,11 +235,25 @@ template <typename T1, size_t PREFIX_LEN>
 }
 } // namespace util
 
+/**
+ * Safer alternative to tfm::format that does not throw string
+ * formatting errors at run-time.
+ *
+ * The `fmt` format string is partially checked at compile-time. At
+ * run-time, if any `tinyformat::format_error` occurs, the error is caught
+ * and the error message is returned instead, so the caller doesn't need
+ * to handle the error.
+ */
 namespace tinyformat {
 template <typename... Args>
-std::string format(util::ConstevalFormatString<sizeof...(Args)> fmt, const Args&... args)
+std::string try_format(util::ConstevalFormatString<sizeof...(Args)> fmt, const Args&... args)
```

Comment:
I renamed it to `try_format` to be clear about the different behaviour compared to `tfm::format`, but agreed that avoiding merge conflicts is useful to coordinate on.

> My preference would be to leave the name as-is and instead just delete the unchecked one

Could you elaborate on this a bit more? We have these 4 other `format` overloads, I don't think you want to just remove them all? At least 2) and 4) would imply touching quite a bit of code to refactor, and it seems not all call sites of `format` are currently constant expressions?

1) `std::string format(const std::string &fmt, const Args&... args)`
2) `std::string format(const char* fmt, const Args&... args)`
3) `void format(std::ostream& out, const char* fmt, const Args&... args)`
4) `bilingual_str format(const bilingual_str& fmt, const Args&... args)`

Reply:
> so `format(const char*, ...)` would have to be deleted for the compiler to pick `format(ConstevalFormatString, ...)`

I think just deleting `format(const char*, ...)` leads to ambiguous overloading?

```
../../../../src/checkqueue.h:136:36: error: call to 'format' is ambiguous
                util::ThreadRename(strprintf("scriptch.%i", n));
                                   ^~~~~~~~~
../../../../src/tinyformat.h:1151:19: note: expanded from macro 'strprintf'
#define strprintf tfm::format
                  ^~~~~~~~~~~
../../../../src/tinyformat.h:1140:13: note: candidate function [with Args = <int>]
std::string format(const std::string &fmt, const Args&... args)
            ^
../../../../src/util/string.h:249:13: note: candidate function [with Args = <int>]
std::string format(util::ConstevalFormatString<sizeof...(Args)> fmt, const Args&... args)
```

> IIUC the remaining `format` taking a `std::string` can be left as-is for now

That (and keeping the `bilingual_str` overload) would be my preference too, but you also asked to remove `./test/lint/lint-format-strings.py` in this pull which I think we shouldn't do before all `tfm::format` overloads are compile-time checked? So that's why I'm a bit confused.

Reply:
Thanks for looking into this further! I was also exploring the concepts route, but hadn't yet resolved all concerns.

1) Re naming: having `format` overloads with significantly differently behaviour (throwing vs non-throwing) didn't quite sit right with me, so I think your `format_raw` (or `format_throw`) suggestion is preferable, and I agree it makes more sense to rename the throwing variant instead of the non-throwing variant (although the downside is that it makes reviewing this PR less straightforward).
2) forcing the `util::ConstevalFormatString` overload for all string literals either requires that:
    a) all callsites include `util/string.h`
    b) `tinyformat.h` includes `util/string.h` (as in your suggestion, although we should probably iwyu export it too), but this introduces a circular dependency
    c) the `util::ConstevalFormatString` overloads are moved to `tinyformat.h`, so the `tinyformat.h` dependency in `util/string.h` can be removed.

I think c) is the preferable approach. It adds more core-specific LoC to tinyformat but I think it is the cleanest overall?

> So an easy fix would be to just call the unchecked c-string functions in bitcoin-cli, which I think is fine.

I agree, although I think I'll clean up the incorrect `bitcoin-cli` format strings here too since that seems like an easy win.